bench/irmin-pack/layers.ml

```diff
@@ -281,13 +302,15 @@ let main () ncommits ncycles depth clear no_freeze show_stats =
   in
   init config;
   let d, _ = Lwt_main.run (with_timer (fun () -> run config)) in
-  let all_commits = ncommits * ncycles in
+  let all_commits = ncommits * (ncycles + 5) in
```

Comment:
Seems odd to always run 5 more cycles than requested. Perhaps we could run `max(5, ncycles)` and update the CLI doc to say that at least 5 runs are necessary?

Reply:
The first 5 cycles are run in addition to the number of cycles passed as arg. 

https://github.com/mirage/irmin/blob/master/bench/irmin-pack/layers.ml#L263 runs the first 5 cycles followed by run_cycles. Maybe we should remove the first_5_cycles and change the doc to say running at least 5 is necessary?

Reply:
Indeed, I think it's better to have a more explicit behaviour.

Out of interest, why is it necessary to have at least 5 cycles? It seems that I might want to just run ~1 cycle when working on the benchmarks, for instance.

Reply:
cc @icristescu :grin:

Reply:
The benchmarks are meant to be somewhat similar to the tezos usecase where every freeze copies the last 5 cycles. So we need to have the first 5 cycles before calling the first freeze. Then each cycle we call freeze again (done by `run_cycles`). You are free to change the benchmarks, just I think the default behaviour should be this one (and more documentation is always welcomed:) )

Reply:
Merge when ready then, @gs0510 :slightly_smiling_face: